lib/sentry/context.ex

```diff
@@ -68,8 +192,52 @@ defmodule Sentry.Context do
     Process.get(@process_dictionary_key) || %{}
   end
 
+  @doc """
+  Adds a new breadcrumb to the `:breadcrumb` context, specific to the current
+  process.
+
+  Breadcrumbs are used to record a series of events that led to a specific
+  instance of an error. Breadcrumbs can contain arbitrary key data to assist in
+  understanding what happened before an error occurred.
+
+  ## Example
+
+      iex> Sentry.Context.add_breadcrumb(message: "first_event")
+      nil
+      iex> Sentry.Context.add_breadcrumb(%{message: "second_event", type: "auth"})
+      %{breadcrumbs: [%{:message => "first_event", "timestamp" => 1562007480}]}
+      iex> Sentry.Context.add_breadcrumb(%{message: "response"})
+      %{
+          breadcrumbs: [
+                %{:message => "second_event", :type => "auth", "timestamp" => 1562007505},
+                %{:message => "first_event", "timestamp" => 1562007480}
+              ]
+      }
+      iex> Sentry.Context.get_all()
+      %{
+          breadcrumbs: [
+                %{:message => "first_event", "timestamp" => 1562007480},
+                %{:message => "second_event", :type => "auth", "timestamp" => 1562007505},
+                %{:message => "response", "timestamp" => 1562007517}
+              ],
+          extra: %{},
+          request: %{},
+          tags: %{},
+          user: %{}
+      }
+  """
+  @spec add_breadcrumb(keyword() | map()) :: nil | map()
   def add_breadcrumb(list) when is_list(list) do
-    add_breadcrumb(Enum.into(list, %{}))
+    if Keyword.keyword?(list) do
+      list
+      |> Enum.into(%{})
+      |> add_breadcrumb
+    else
+      raise ArgumentError, """
+      Sentry.Context.add_breadcrumb/1 only accepts keyword lists or maps.
+      Received a non-keyword list.
+      """
```

Comment:
Noticed that technically this function accepts a list of any sort. If given one it raises an argument error in the internals of `:maps.from_list` like so:
```
iex(1)> Sentry.Context.add_breadcrumb(["dog"])
** (ArgumentError) argument error
    (stdlib) :maps.from_list(["dog"])
    (elixir) lib/map.ex:174: Map.new/1
    (sentry) lib/sentry/context.ex:225: Sentry.Context.add_breadcrumb/1``
```
Hopefully this change helps someone in the future!

This does technically introduce a tiny performance hit, but unless a breadcrumb has thousands of keys (since `Keyword.keyword?/1` traverses the entire list) it should be negligible.

Feel free to tell me this isn't worth it!